query/sources/impl/src/test/groovy/org/codice/ddf/admin/sources/test/SourceTestCommons.groovy

```diff
@@ -74,62 +76,74 @@ class SourceTestCommons {
+                    (FACTORY_PID_KEY): F_PID,
+                    (SERVICE_PID_KEY): S_PID_1,
+                    (USERNAME)       : TEST_USERNAME
+            ],
+            (S_PID_2): [
+                    (PASSWORD)       : TEST_PASSWORD,
+                    (ID)             : SOURCE_ID_2,
+                    (FACTORY_PID_KEY): F_PID,
+                    (SERVICE_PID_KEY): S_PID_2,
+                    (USERNAME)       : TEST_USERNAME
+            ]
+    ]
+
+    static configToBeDeleted = [
             (PASSWORD)       : TEST_PASSWORD,
             (ID)             : SOURCE_ID_1,
             (FACTORY_PID_KEY): F_PID,
-            (SERVICE_PID_KEY): S_PID_1,
+            (SERVICE_PID_KEY): S_PID,
             (USERNAME)       : TEST_USERNAME
-        ],
-        (S_PID_2): [
-            (PASSWORD)       : TEST_PASSWORD,
-            (ID)             : SOURCE_ID_2,
-            (FACTORY_PID_KEY): F_PID,
-            (SERVICE_PID_KEY): S_PID_2,
-            (USERNAME)       : TEST_USERNAME
-        ]
     ]
 
-    static configToBeDeleted = [
-        (PASSWORD)                    : TEST_PASSWORD,
-        (ID)                          : SOURCE_ID_1,
-        (FACTORY_PID_KEY)             : F_PID,
-        (SERVICE_PID_KEY)             : S_PID,
-        (USERNAME)                    : TEST_USERNAME
-    ]
+    /**
+     * Needed to mock out the SecureCxfClientFactory. Since it is an embedded class we are just going to override
+     * the method used to create the factory and substitute our own. We also need to mock out the endpointIsReachable
+     * method since it tries to make network requests.
+     */
+    static class TestRequestUtils extends RequestUtils {
+
+        SecureCxfClientFactory factory
+
+        boolean endpointIsReachable
 
-    static createResponseFieldResult(boolean hasError, String responseBody, int code, String requestUrl) {
-        def result = new ReportWithResultImpl<ResponseField>()
-        if (hasError) {
-            result.addArgumentMessage(new ErrorMessageImpl('TEST_CODE', ['some', 'path']))
-        } else {
-            ResponseField responseField = new ResponseField()
-            responseField.responseBody(responseBody)
-            responseField.statusCode(code)
-            responseField.requestUrl(requestUrl)
-            result.result(responseField)
+        TestRequestUtils(SecureCxfClientFactory mockFactory, boolean isReachable) {
+            factory = mockFactory
+            endpointIsReachable = isReachable
         }
 
-        return result
+        @Override
+        protected SecureCxfClientFactory<WebClient> createFactory(String url, CredentialsField creds) {
+            return factory
+        }
+
+        @Override
+        public ReportImpl endpointIsReachable(UrlField urlField) {
+            def report = new ReportImpl()
+            if (!endpointIsReachable) {
+                report.addArgumentMessage(new ErrorMessageImpl("some code"))
```

Comment:
Should we use the some constant for the error code? Something like MOCK_REQUEST_ERROR_CODE?